src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -228,7 +250,7 @@ public ZooKeeperSaslClient getSaslClient() {
      * the public methods will not be exposed as part of the ZooKeeper client
      * API.
      */
-    static class ZKWatchManager implements ClientWatchManager {
+    class ZKWatchManager implements ClientWatchManager {
```

Comment:
I made it non-static to use non-static variable form outer class.
 shall i change ZKWatchManager constructor to pass disableAutoWatchReset? so there will not be any need to access outer class variable